components-core/src/test/java/com/adyen/checkout/components/core/internal/analytics/DefaultAnalyticsManagerTest.kt

```diff
@@ -0,0 +1,160 @@
+@ExtendWith(MockitoExtension::class, LoggingExtension::class, TestDispatcherExtension::class)
+internal class DefaultAnalyticsManagerTest(
+    @Mock private val analyticsRepository: NewAnalyticsRepository,
+) {
+
+    private lateinit var analyticsManager: DefaultAnalyticsManager
+
+    @BeforeEach
+    fun setup() = runTest {
+        analyticsManager = createAnalyticsManager()
+    }
+
+    @Nested
+    @DisplayName("when initializing and")
+    inner class InitializeTest {
+
+        @Test
+        fun `sending events is disabled, then checkoutAttemptId is anonymous`() = runTest {
+            analyticsManager = createAnalyticsManager(AnalyticsParamsLevel.NONE)
+
+            analyticsManager.initialize(this@InitializeTest, this)
+
+            assertEquals(
+                DefaultAnalyticsManager.CHECKOUT_ATTEMPT_ID_FOR_DISABLED_ANALYTICS,
+                analyticsManager.getCheckoutAttemptId(),
+            )
+            verify(analyticsRepository, never()).fetchCheckoutAttemptId()
+        }
+
+        @Test
+        fun `fetching checkoutAttemptId succeeds, then it is set`() = runTest {
+            whenever(analyticsRepository.fetchCheckoutAttemptId()) doReturn "test value"
+
+            analyticsManager.initialize(this@InitializeTest, this)
+
+            assertEquals("test value", analyticsManager.getCheckoutAttemptId())
+            analyticsManager.clear(this@InitializeTest)
+        }
+
+        @Test
+        fun `fetching checkoutAttemptId fails, then checkoutAttemptId is null`() = runTest {
+            whenever(analyticsRepository.fetchCheckoutAttemptId()) doAnswer { error("test") }
+
+            analyticsManager.initialize(this@InitializeTest, this)
+
+            assertNull(analyticsManager.getCheckoutAttemptId())
+        }
+
+        @Test
+        fun `initialize is called twice, then the second time is ignored`() = runTest {
+            whenever(analyticsRepository.fetchCheckoutAttemptId()) doAnswer { error("test") }
+
+            analyticsManager.initialize(this@InitializeTest, this)
+            analyticsManager.initialize(this@InitializeTest, this)
+
+            verify(analyticsRepository, times(1)).fetchCheckoutAttemptId()
+        }
+    }
+
+    @Nested
+    @DisplayName("when tracking events and")
+    inner class TrackEventTest {
+
+        @Test
+        fun `sending events is disabled, then events should not be stored`() = runTest {
+            analyticsManager = createAnalyticsManager(AnalyticsParamsLevel.NONE)
+            analyticsManager.initialize(this@TrackEventTest, this)
+
+            analyticsManager.trackEvent(GenericEvents.rendered("dropin", false))
+
+            verify(analyticsRepository, never()).storeEvent(any())
+        }
+
+        @Test
+        fun `sending events is enabled, then events should be stored`() = runTest {
+            analyticsManager.initialize(this@TrackEventTest, this)
+            val event = GenericEvents.rendered("dropin", false)
```

Comment:
What do you think of creating a new `AnalyticsEvent` internally for the usage of this test?

This way we can explicitly set the `shouldForceSend` flag and the tests will not fail if the implementation of `Info` or `Log` changes or any specific event changes its type.